earth_enterprise/src/common/geFilePool.cpp

```diff
@@ -20,17 +20,65 @@
 #include <third_party/rsa_md5/crc32.h>
 #include <khEndian.h>
 
+// ****************************************************************************
+// ***  POSIXFileAcessor
+// ****************************************************************************
+int POSIXFileAccessor::Open(const std::string &fname, int flags, mode_t createMask) {
+  return khOpen(fname, flags, createMask);
+}
+
+int POSIXFileAccessor::FsyncAndClose(std::shared_ptr<AbstractFileIdentifier> pID) {
+  return khFsyncAndClose(pID->getAsFD());
+}
+
+int POSIXFileAccessor::Close(std::shared_ptr<AbstractFileIdentifier> pID) {
+  return khClose(pID->getAsFD());
+}
+
+bool POSIXFileAccessor::PreadAll(std::shared_ptr<AbstractFileIdentifier> pID, void* buffer, size_t size, off64_t offset) {
+  return khPreadAll(pID->getAsFD(), buffer, size, offset);
+}
+
+bool POSIXFileAccessor::PwriteAll(std::shared_ptr<AbstractFileIdentifier> pID, const void* buffer, size_t size, off64_t offset) {
+  return khPwriteAll(pID->getAsFD(), buffer, size, offset);
+}
+
+// ****************************************************************************
+// ***  FileIdentifierFactory
+// ****************************************************************************
+namespace FileIdentifierFactory {
+  static std::shared_ptr<AbstractFileIdentifier> getIdentifier(int i) {
+    return std::make_shared<POSIXFileIdentifier>(i);
+  }
+}
+
+// ****************************************************************************
+// ***  FileAccessorFactory
+// ****************************************************************************
+namespace FileAccessorFactory {
+  static std::shared_ptr<AbstractFileAccessor> getAccessor(const std::string &fname) {
+    return std::make_shared<POSIXFileAccessor>();
```

Comment:
I'm not sure if creating a new file accessor every time is the right approach. That could result in a lot of memory allocation and clean up. It might be cleaner to treat this more like a singleton where it always returns the same file accessor every time.

Reply:
We discussed that briefly. We weren't sure about the implications if multiple files were being accessed by the system at one time.